src/main/kotlin/com/hibob/academy/employeeFeedback/service/FeedbackService.kt

```diff
@@ -0,0 +1,43 @@
+package com.hibob.academy.employeeFeedback.service
+
+import com.hibob.academy.employeeFeedback.dao.*
+import jakarta.ws.rs.BadRequestException
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackService(private val feedbackDao: FeedbackDao, private val employeeDao: EmployeeDao) {
+    fun submitFeedback(employeeId: Int, companyId: Int, anonymity: AnonymityType, feedback: String): Boolean {
+        var employeeIdForFeedback: Int? = employeeId
+        if (anonymity == AnonymityType.ANONYMOUS)
+            employeeIdForFeedback = null
+
+        if (feedback.length < 30 || feedback.length > 500)
+            return false
+
+        val feedbackForSubmission =
+            FeedbackForSubmission(employeeIdForFeedback, companyId, anonymity, feedback)
+        feedbackDao.submitFeedback(feedbackForSubmission)
+        return true
+    }
+
+    fun viewAllSubmittedFeedback(employeeId: Int, companyId: Int): List<FeedbackInfo> {
+        return feedbackDao.viewAllSubmittedFeedback(companyId)
+    }
+
+    fun viewStatusOfMyFeedback(employeeId: Int, companyId: Int, feedbackId: Int): Map<Int, Boolean> {
+        val feedbackInfo = feedbackDao.getFeedbackById(feedbackId, companyId)
+        feedbackInfo?.let{
+            val idOfEmployeeThatWroteFeedbackDao = feedbackInfo.employeeId
+                ?: throw Exception("Can't check status of anonymous feedback")
```

Comment:
you are throwing a general exception here, does it make sense?
think of how you will handle this in the calling method?

What do you want the user to recieve in this case?

Reply:
I want the user to receive a Response that says that it cant check status of an anonymous feedback